OpenRA.Mods.RA2/Traits/ChronoDelivery.cs

```diff
@@ -0,0 +1,105 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System;
+using OpenRA.Mods.Common.Activities;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Activities;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.RA2.Traits
+{
+	public class ChronoDeliveryInfo : ITraitInfo, Requires<HarvesterInfo>
+	{
+		[Desc("The number of ticks between each check to see if we can teleport home.")]
+		public readonly int CheckTeleportDelay = 10;
+
+		public virtual object Create(ActorInitializer init) { return new ChronoDelivery(init.Self, this); }
+	}
+
+	public class ChronoDelivery : INotifyHarvesterAction, ITick
```

Comment:
Please rename this (and therefor the Info) to `ChronoResourceDelivery` just to be a bit more verbose.